web/vibe/web/rest.d

```diff
@@ -1447,7 +1448,7 @@ private HTTPServerRequestDelegate jsonMethodHandler(alias Func, size_t ridx, T)(
 	import vibe.textfilter.urlencode : urlDecode;
 
 	enum Method = __traits(identifier, Func);
-	alias PTypes = ParameterTypeTuple!Func;
+	alias PTypes = staticMap!(Unqual, ParameterTypeTuple!Func);
```

Comment:
This should come with a comment. From my understanding:

```suggestion
    // We need mutable types for deserialization
	alias PTypes = staticMap!(Unqual, ParameterTypeTuple!Func);
```

However there should probably be a check for immutable or the error messages aren't going to be great.

Reply:
immutable params compile fine tho?

Reply:
So, mutable does not implicitly converts to its `immutable` counterpart, unless the type has no indirection.
Which means that, by deserializing things as `T`, we can't later bind it to a parameter of type `immutable(T)`.
And currently, IIUC, the error is that we try to modify an `immutable(T)` during deserialization.
If it's not one, it has to be the other. If it's neither, two possibilities:
- We are deserializing in one go and we shouldn't have the error to begin with (or the fix is even more trivial?);
- There is a cast somewhere that breaks the type system;